portal/static/js/main.js

```diff
@@ -4875,106 +909,95 @@ var tnthAjax = {
-                                        default:
-                                            found = true; //default is to return both suspended and consented entries
-                                    };
+                                    switch (filterStatus) {
+                                    case "suspended":
+                                        if (suspended) found = true;
+                                        break;
+                                    case "purged":
+                                        found = true;
+                                        break;
+                                    case "consented":
+                                        if (!suspended) {
+                                            if (item.staff_editable && item.send_reminders && item.include_in_reports) found = true;
+                                        }
+                                        break;
+                                    default:
+                                        found = true; //default is to return both suspended and consented entries
+                                    }
                                     if (found) consentedOrgIds.push(orgId);
 
-                                };
-                            };
+                                }
+                            }
                         }
-                    };
+                    }
 
                 } else {
                     return false;
                 }
-            };
+            }
         });
         return consentedOrgIds.length > 0 ? consentedOrgIds : null;
     },
     removeObsoleteConsent: function() {
-        var userId = $("#fillOrgs").attr("userId");
-        var co = [];
-        var OT = this.getOrgTool();
+        var userId = $("#fillOrgs").attr("userId"), co = [], OT = this.getOrgTool();
         $("#userOrgs input[name='organization']").each(function() {
             if ($(this).is(":checked")) {
                 var po = OT.getElementParentOrg(this);
                 co.push($(this).val());
-                if (hasValue(po)) co.push(po);
-            };
+                if (po) co.push(po);
+            }
         });
-        //exclude currently selected orgs
-        tnthAjax.deleteConsent(userId, {org: "all", exclude: co.join(",")});
+        tnthAjax.deleteConsent(userId, {org: "all", exclude: co.join(",")}); //exclude currently selected orgs
     },
     handleConsent: function(obj) {
-        var self = this;
-        var OT = this.getOrgTool();
-        var userId = $("#fillOrgs").attr("userId");
-        if (!hasValue(userId)) {
-            userId = $("#userOrgs").attr("userId");
-        }
+        var self = this, OT = this.getOrgTool(), userId = $("#fillOrgs").attr("userId") || $("#userOrgs").attr("userId");
         var configVar = $("#profile_CONSENT_WITH_TOP_LEVEL_ORG").val();
-        if (!hasValue(configVar)) {
+        if (!configVar) {
```

Comment:
You might want to invert this, checking localeStorage should be quicker than checking the DOM